defender/wallet-migrator/approveTokens.js

```diff
@@ -0,0 +1,143 @@
+require('dotenv').config();
+const yaml = require('js-yaml');
+const fs = require('fs');
+const axios = require('axios');
+const { ethers } = require('ethers');
+const { RelayClient } = require('defender-relay-client');
+// grab secrets from .env file
+const GOERLI_RPC_URL = process.env.GOERLI_RPC_URL;
+const MAINNET_RPC_URL = process.env.MAINNET_RPC_URL;
+const PRIVATE_KEY = process.env.PRIVATE_KEY;
+// grab Defender API Key, Secret Key and Covalent API Key from .secrets/dev.yml file
+// we need to replace the "-" from yaml files, or it throws an error when importing
+let secretsFile = yaml.load(fs.readFileSync('defender/.secrets/dev.yml', 'utf8', { schema: 'JSON_SCHEMA' }));
+secretsFile = JSON.parse(JSON.stringify(secretsFile).replace(/-/g, ''));
+const DEFENDER_API_KEY = secretsFile.keys.defenderapikey;
+const DEFENDER_API_SECRET = secretsFile.keys.defenderapisecret;
+const COVALENT_API_KEY = secretsFile.keys.covalentapikey;
+
+// grab network from config file
+const configFile = yaml.load(fs.readFileSync('defender/wallet-migrator/config.dev.yml', 'utf8'));
+const network = configFile.network;
+
+const erc20Abi = [
+  'function approve(address _spender, uint256 _value) external',
+  'function allowance(address owner, address spender) external view returns (uint256)',
+];
+
+const erc721Abi = [
+  'function approve(address spender, uint256 amount) external returns (bool)',
+  'function setApprovalForAll(address operator, bool approved) public',
+  'function isApprovedForAll(address owner, address operator) public view returns (bool)',
+];
+
+// returns axios get request to retrieve wallet balance of erc20 and erc721 tokens
+async function get(url, method, headers, auth) {
+  return axios({
+    url,
+    method,
+    headers,
+    auth,
+  });
+}
+
+async function main() {
+  // initiate a Defender Relay Client to get address of Relayer
+  const relayClient = new RelayClient({ apiKey: DEFENDER_API_KEY, apiSecret: DEFENDER_API_SECRET });
+  const relayerInfo = await relayClient.list();
+  let relayerAddress;
+  // a user might have more than 1 relayer in their account, so we need to get the one for this stack
+  relayerInfo.items.forEach(relayer => {
+    if (relayer.name === 'Migrator Relay' && relayer.stackResourceId === 'wallet_migrator.relayer-1') {
+      relayerAddress = relayer.address;
+    }
+  });
+
+  let rpcEndpoint;
+  let chainId;
+
+  if (network === 'goerli') {
+    rpcEndpoint = GOERLI_RPC_URL;
+    chainId = '5';
+  } else if (network === 'mainnet') {
+    rpcEndpoint = MAINNET_RPC_URL;
+    chainId = '1';
+  }
```

Comment:
`provider` and `chainId` can be simplified and support any* netowrk if RelayClientProvider and Signer are used instead of `RelayClient` see README comment and code.